core/mutex.class.inc.php

```diff
@@ -258,6 +258,38 @@ public function InitMySQLSession()
 		{
 			throw new Exception("Could not connect to the DB server (host=$sServer, user=$sUser): ".mysqli_connect_error().' (mysql errno: '.mysqli_connect_errno().')');
 		}
+
+		// Make sure that the server variable wait_timeout is at least 86400 seconds for this connection,
+		// since the lock will be released if/when the connection times out.
+
+		// BEWARE: If you want to check the value of this variable, when run from an interactive console "SHOW VARIABLES LIKE 'wait_timeout'" actually
+		// returns the value of the variable 'interactive_timeout' which may be quite different.
+
+		$sSql = "SHOW VARIABLES LIKE 'wait_timeout'";
+		$result = mysqli_query($this->hDBLink, $sSql);
+		if (!$result)
+		{
+		    throw new Exception("Failed to issue MySQL query '".$sSql."': ".mysqli_error($this->hDBLink).' (mysql errno: '.mysqli_errno($this->hDBLink).')');
+		}
+		if ($aRow = mysqli_fetch_array($result, MYSQLI_BOTH))
```

Comment:
Using the default (both) while using only numeric value.
```suggestion
		if ($aRow = mysqli_fetch_array($result, MYSQLI_NUM))
```

Reply:
Thanks for the suggestion, but I'm not absolutly sure of the real benefits, and as there was already manual tests done with the current code I'm worried changing this could cause regressions : I'm keeping code like this for now.

Reply:
Memory optimisation: `$aRow`
before:
```
array(2) {
  [0]=>
  string(5) "12345"
  ["wait_timeout"]=>
  string(5) "12345"
}
```
after:
```
array(1) {
  [0]=>
  string(5) "12345"
}
```

See [docs](https://www.php.net/manual/en/mysqli-result.fetch-array.php)

Reply:
Thanks !
Keeping this in mind for next similar operations